### PR TITLE
Add JsonWriter.jsonValue API

### PR DESCRIPTION
**Pull request by JakeWharton**
This is symmetric with JsonReader.readJsonValue in that it writes an Object comprised of maps, lists, and some scalars.

### CODE REVIEW

File: moshi/src/main/java/com/squareup/moshi/JsonWriter.java [hunk 1]
```diff
@@ -386,6 +388,57 @@ public final JsonWriter value(BufferedSource source) throws IOException {
...
+   * @return this writer.
+   * @see JsonReader#readJsonValue()
+   */
+  public final JsonWriter jsonValue(@Nullable Object value) throws IOException {
+    if (value instanceof Map<?, ?>) {
+      beginObject();
+      for (Map.Entry<?, ?> entry : ((Map<?, ?>) value).entrySet()) {
+        Object key = entry.getKey();
+        if (!(key instanceof String)) {
+          throw new IllegalArgumentException(
```
Comment (swankjesse) [inline moshi/src/main/java/com/squareup/moshi/JsonWriter.java]: what does this do if key is null?

Comment (JakeWharton) [reply moshi/src/main/java/com/squareup/moshi/JsonWriter.java]: NPE creating message. Will fix and test later today. Should also test arrays and maps with null values.
